src/Streamly/Internal/Data/Parser/ParserD.hs

```diff
@@ -408,25 +390,26 @@ takeGE cnt (Fold fstep finitial fextract) = Parser step initial extract
 {-# INLINE takeWhile #-}
 takeWhile :: Monad m => (a -> Bool) -> Fold m a b -> Parser m a b
 takeWhile predicate (Fold fstep finitial fextract) =
-    Parser step initial extract
+    Parser step initial fextract
 
     where
 
-    initial = finitial
+    initial = do
```

Comment:
`take`, `takeGE`, `takeWhile`. Were the changes made to simplify the code or was the previous implementation incorrect?

Reply:
Oh, is this because we'll remove this anyway once we stepify the parser initial?

Reply:
Yes, I do not want to complicate the code when we know we have to revert that soon. It will be double effort.